apps/mobile/src/screens/connection.tsx

```diff
@@ -48,27 +50,31 @@ export function ConnectionScreen({
     <Stack.Navigator initialRouteName={initialRoute}>
       <Stack.Screen
         name="index"
-        component={RiseScreen}
-        initialParams={{
-          id: route.params.id,
-          path: connection?.path || '',
-        }}
         options={{
           headerLeft: BackButton,
           title: connection?.label || connection?.path,
         }}
-      />
+      >
+        {() => <RiseScreen connection={connection!} path={connection!.path || ''} />}
+      </Stack.Screen>
       <Stack.Screen
         name="rise"
-        component={RiseScreen}
         getId={({ params }) => params.path}
-        initialParams={{
-          id: route.params.id,
-        }}
         options={({ route }) => ({
           title: route.params.path,
         })}
-      />
+      >
+        {({ route }) => <RiseScreen connection={connection!} path={route.params.path} />}
```

Comment:
Did a small consulting session with Satya regarding TypeScript and the way we manage params and decided to change from "Component" and "initialParams" to [passing params explicitly as per the documentation](https://reactnavigation.org/docs/hello-react-navigation/#passing-additional-props). 

This makes our code much cleaner and allows us to write `navigate('qr-code')` from within BackButton without having to read and pass connectionId, that we already know anyway (in context of current navigation stack).

Not saying this is final pattern, but it's definitely much cleaner and gets the job done!